lib/ansible/modules/cloud/amazon/emr_cluster.py

```diff
@@ -0,0 +1,363 @@
+      ec2_subnet_id: subnet-923a31db
+      configurations:
+        - classification: spark
+          properties:
+            maximizeResourceAllication: true
+      additional_slave_security_groups:
+        - sg-8eb2b1f1
+      additional_master_security_groups:
+        - sg-8eb2b1f1
+      instance_groups:
+        - name: master
+          instance_role: MASTER
+          instance_count: 1
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+        - name: core
+          instance_role: CORE
+          instance_count: 4
+          instance_type: m3.xlarge
+          market: ON_DEMAND
+'''
+
+RETURN = '''
+# For more information see http://boto3.readthedocs.io/en/latest/reference/services/emr.html#EMR.Client.describe_cluster
+---
+cluster:
+  description: dictionary containing informations about EMR Cluster
+  returned: success
+  type: complex
+  contains:
+    id:
+      description: the unique id for the cluster
+      type: str
+      sample: j-2JNVMF38MZVHT
+    name:
+      description: the name of the cluster
+      type: str
+      sample: emr-jobs
+    status:
+      description: the current status defailts about the cluster
+      type: dict
+    ec2_instaces_attributes:
+      description: provides information about ec2
+      type: dict
+    log_uri:
+      description: path for logs location in S3
+      type: str
+    release_label:
+      description: the release label for the Amazon EMR
+      type: str
+    auto_terminate:
+      description: if true cluster should terminate after completing all step
+      type: boolean
+    applications:
+      description: list of the applications installed on this cluster
+      type: list
+    tags:
+      description: list of tags
+      type: dict
+    service_role:
+      description: IAM role that will be assumed by the Amazon AEMR service
+      type: str
+    master_public_dns_name:
+      description: the dns name of the master node
+      type: str
+    configurations:
+      description: the list of configurations supplied to EMR cluster
+      type: dict
+'''
+
+try:
+    from botocore.exception import ClientError
+except ImportError:
+    pass  # will be picked up from imported HAS_BOTO3
+
+import re
+import traceback
+import time
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, boto3_conn, get_aws_connection_info
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict, snake_dict_to_camel_dict, ansible_dict_to_boto3_tag_list
+
+
+def terminate_emr_cluster(client, module):
+    cluster_id = module.params.get('cluster_id')
+    wait = module.params.get('wait')
+    wait_timeout = module.params.get('wait_timeout')
+
+    if not cluster_id:
+        module.fail_json(msg='cluster_id required for state absent')
+
+    client.terminate_job_flows(JobFlowIds=[cluster_id])
+
+    resource = client.describe_cluster(ClusterId=cluster_id)['Cluster']
+
+    if wait:
```

Comment:
There is a built-in waiter you can use for this: http://boto3.readthedocs.io/en/latest/reference/services/emr.html#EMR.Waiter.ClusterTerminated

You can modify the waiter config to wait the correct amount of time by adding the WaiterConfig keyword:
```
waiter.wait(
    ClusterId=cluster_id,
    WaiterConfig={'Delay': 5, 'MaxAttempts': wait_timeout // 5}
)['Cluster']
```